src/main/webapp/mapScript.js

```diff
@@ -168,23 +125,40 @@ function drawRoute() {
     lng: markersArray[markersArray.length - 1].data.lng,
   });
 
-  let waypoints = [];
+  var waypoints = [];
   for (var i = 1; i < markersArray.length - 1; i++) {
-    waypoints.push(markersArray[i].marker);
+    let waypointLatLng = new google.maps.LatLng({
```

Comment:
markersArray[i].marker already contain Marker variable so there are actually no reason in using markersArray[i].data.